### ISSUE TITLE
Error transforming TSX

### ISSUE THREAD

**Issue by TheBosZ** [Rich-Harris/tippex#12]
I'm using Rollup with the Typescript plugin for our React app.

I get an error when I use JS inside a JSX block.

```
import * as React from 'react';
import { connect } from 'react-redux';

interface HomeProps {
    gateways: any;
}

class Home extends React.Component<HomeProps, any> {
    public render () {
        const gateways = this.props.gateways;
        return (
            <div>
                <div>This is Home</div>
                <ul>
                    {gateways.map((gateway) =>
                    <li>{gateway}</li>
                    )}
                </ul>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    gateways: state.APIClientReducer.gateways
});

export default connect(mapStateToProps)(Home);
```

Causes this error:

```
Error transforming C:/projects/cloud-iris-web-frontend/src/components/Home/Home.
tsx: rollup-plugin-typescript: Unexpected character (22:1). If this is valid Jav
aScript, it's probably a bug in tippex. Please raise an issue at https://github.
com/Rich-Harris/tippex/issues - thanks!


}const mapStateToProps = state => ({

  ^; when processing: 'C:/projects/cloud-iris-web-frontend/src/components/Home/H
ome.tsx'
Error: Error transforming C:/projects/cloud-iris-web-frontend/src/components/Hom
e/Home.tsx: rollup-plugin-typescript: Unexpected character (22:1). If this is va
lid JavaScript, it's probably a bug in tippex. Please raise an issue at https://
github.com/Rich-Harris/tippex/issues - thanks!


}const mapStateToProps = state => ({

  ^; when processing: 'C:/projects/cloud-iris-web-frontend/src/components/Home/H
ome.tsx'
    at getErasedCode (C:\projects\cloud-iris-web-frontend\node_modules\rollup-pl
ugin-typescript\dist\rollup-plugin-typescript.cjs.js:144:9)
    at fix (C:\projects\cloud-iris-web-frontend\node_modules\rollup-plugin-types
cript\dist\rollup-plugin-typescript.cjs.js:113:18)
    at Object.transform (C:\projects\cloud-iris-web-frontend\node_modules\rollup
-plugin-typescript\dist\rollup-plugin-typescript.cjs.js:271:50)
    at C:\projects\cloud-iris-web-frontend\node_modules\rollup\src\utils\transfo
rm.js:19:35
```

**Comment by ochafik** [comment]
Minimal repro that can be pasted in tippex's test.js:

``` javascript
it( 'does not crash with "Unexpected character"', () => {
  tippex.erase('<div></div>; function x() { <div></div> } \n');
});
```

Crash:

```
Error: Unexpected character (1:41). If this is valid JavaScript, it's probably a bug in tippex. Please raise an issue at https://github.com/Rich-Harris/tippex/issues – thanks!

<div></div>; function x() { <div></div> } 

                                         ^
```

Note that getting rid of that space... causes a different crash:

``` javascript
it( "does not crash with 'Cannot read property '0' of null'", () => {
  tippex.erase('<div></div>; function x() { <div></div> }\n');
});
```

Crashes with:

```
     TypeError: Cannot read property '0' of null
      at find (dist/tippex.umd.js:184:45)
      at Object.erase (src/index.js:223:9)
      at Context.it (test/test.js:151:25)
```

**Comment by Rich-Harris** [comment]
Fixed in 2.3.0 – thanks
